geoserver/src/eumetsat/pinning/src/main/java/org/geoserver/eumetsat/pinning/ViewsEvaluator.java

```diff
@@ -0,0 +1,543 @@
+
+                    ViewRecord record =
+                            new ViewRecord(id, timeOriginal, timeMain, layersList, lastUpdated);
+                    logger.log(Level.FINER, record);
+                    return record;
+                }
+            }
+        }
+        return null;
+    }
+
+    private void setPinLayer(Instant originalTime, Instant mainTime, MappedLayer layer, boolean add)
+            throws SQLException, IOException {
+
+        boolean mainGreaterThanOriginal = mainTime.isAfter(originalTime);
+        Instant minTime = mainGreaterThanOriginal ? originalTime : mainTime;
+        Instant maxTime = mainGreaterThanOriginal ? mainTime : originalTime;
+        Instant start = layer.getNearest(getLeft(minTime));
+        Instant end = layer.getNearest(getRight(maxTime));
+        setPinLayerRange(start, end, layer, add);
+    }
+
+    private void setPinLayerRange(Instant start, Instant end, MappedLayer layer, boolean add)
+            throws SQLException {
+        logger.log(
+                Level.FINE,
+                String.format(
+                        "%sinning layer %s in range (%s,%s)",
+                        add ? "P" : "Unp", layer.getGeoServerLayerIdentifier(), start, end));
+        String updateQuery =
+                pinBatcher.update(
+                        layer.getTableName(), layer.getTemporalAttribute(), start, end, add);
+        logger.log(Level.FINEST, "pinning query:" + updateQuery);
+    }
+
+    private Instant getLeft(Instant time) {
+        return time.minus(duration);
+    }
+
+    private Instant getRight(Instant time) {
+        return time.plus(duration);
+    }
+
+    private void insertView(ViewRecord view) throws SQLException {
+        logger.log(Level.FINE, "Inserting the view in the views table: " + view.getId());
+        String insertQuery = viewsBatcher.insertView(view);
+        logger.log(Level.FINEST, "Inserting view query: " + insertQuery);
+    }
+
+    private void updateView(ViewRecord view) throws SQLException {
+        logger.log(Level.FINE, "Updating the view in the views table: " + view.getId());
+        String updateQuery = viewsBatcher.updateView(view);
+        logger.log(Level.FINEST, "Updating view query: " + updateQuery);
+    }
+
+    private void deleteView(Long viewId) throws Exception {
+        logger.log(Level.FINE, "Deleting the view from the views table: " + viewId);
+        String deleteQuery = viewsBatcher.deleteView(viewId);
+        logger.log(Level.FINEST, "Deleting view query: " + deleteQuery);
+    }
+
+    private void resetPin(MappedLayer layer) throws SQLException {
+        String tableName = layer.getTableName();
+        logger.log(Level.FINE, String.format("Resetting pin for table %s", tableName));
+        String resetQuery = pinBatcher.resetPins(tableName);
+        logger.log(Level.FINEST, "Resetting pin query: " + resetQuery);
+    }
+
+    private void pinGeoserverLayer(
```

Comment:
The action says "pin", but then it can be used for unpin as well. Maybe call it "updateLayerPins" instead?